### ISSUE TITLE
Explosions Don't Always Deal Appropriate Damage to Large Targets

### ISSUE THREAD

**Issue by Jlobblet** [Regalis11/Barotrauma#3536]
- [x] I have searched the issue tracker to check if the issue has already been reported.

**Description**
Explosions on large creatures will frequently not do damage, or do far less damage than anticipated. This makes railguns much weaker than intended. Observed mostly with large creatures such as moloch or ITA endowrm, where only the impact damage from the shell was recorded.

**Steps To Reproduce**
1. Start a round in a sub with a railgun.
2. Load explosives into the railgun.
3. Spawn a moloch and shoot it.
4. Observe damage.

**Version**
0.10.4.0
Windows 10 1903

**Comment by Ek-Videogames** [comment]
_Yes, what seems to be the bug?_

The viability of railguns in general, and especially nukes suffers because of this. Just for reference, you can actually kill endworms far more easily with piercing coilguns than you can with an unlimited number of nuclear shells.

_- also there was already a ticket for this a long time ago, but it has been lost to time and this is still a good reason anyways._

**Comment by Regalis11** [comment]
> I once suggested a potential idea for a fix, in which an monster's body "radius" is subtracted from the explosion distance.

This is actually how we handle it now:
```
float limbRadius = limb.body.GetMaxExtent();
dist = Math.Max(0.0f, dist - ConvertUnits.ToDisplayUnits(limbRadius));
```

So I'd guess the issue somewhere else, need to take a closer look into this.

**Comment by Regalis11** [comment]
Fixed in https://github.com/Regalis11/Barotrauma-development/commit/dcbdb1bcd6caeab65e2ea1807c85f33d37347025.

This happened because the explosion damage is divided by the number of limbs the character has. The intention behind it was to prevent characters with more limbs taking more damage from explosions. It didn't work too well on the large monsters, because the explosion would often only hit one or two of the character's limbs, but the damage would still get divided by the total number of limbs.

**Comment by itchyOwl** [comment]
The issue was not only with explosions. While trying to make the railgun shells more powerful, I found that it didn't make any difference whether the shells do 100 or 10000 dmg. And it was not due to explosions, because normal railgun shells don't even explode. Rather the issue was more generally in how the damage was applied. Before applying the damage, we instantiate afflictions, but afflictions cannot have more strength than the max strength (defined as 100 for most damage types). Hence, the nuclear shell definition for 500 burn damage was actually clamped to 100. And THEN the issue with exposions would make things worse (because the damage was further divided by the limb count). So, in practice, previously it didn't matter if an affliction was defined to give more than 100 damage, the damage was always 100 at max per affliction. This issue is fixed in Regalis11/Barotrauma-development@a764608. In consequence, if there are any attacks defined to make more than 100 damage, they will now be stronger. Something to consider in your mods.

**Comment by Rokvach** [comment]
Based of feedback this is now working correctly, closing.
